src/an_sso.coffee

```diff
@@ -206,6 +208,35 @@ window.AsmodeeNet = (->
                     else
                         window.location = '/'
 
+    baseLinkAction = (endpoint, options) ->
+        options = options || {}
+        iFrame.saveOptions = AsmodeeNet.extend {}, options if settings.display == 'iframe'
+        state = getCryptoValue()
+        nonce = getCryptoValue()
+        setItem('state', state, if settings.display == 'iframe' then 1440 else 20)
+        setItem('nonce', nonce, if settings.display == 'iframe' then 1440 else 20)
+        settings.callback_signin_success = options.success || settings.callback_signin_success
+        settings.callback_signin_error = options.error || settings.callback_signin_error
+        options.path = endpoint +
+            '?display=' + settings.display +
+            '&response_type=' + encodeURI(settings.response_type) +
+            '&state=' + state +
+            '&client_id=' + settings.client_id +
+            '&scope=' + settings.scope
+        options.path += '&redirect_uri=' + encodeURI(settings.redirect_uri) if settings.redirect_uri
+        options.path += '&nonce='+nonce if settings.response_type.search('id_token') >= 0
+        if Object.keys(settings.display_options).length > 0
```

Comment:
Is testing for the presence of at least one key necessary? If `settings.display_options` is empty, the loop will never be run, will it?